x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/advanced_editor.tsx

```diff
@@ -0,0 +1,299 @@
+import { i18n } from '@kbn/i18n';
+import { useDebounce } from 'react-use';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiIcon,
+  EuiFieldNumber,
+  EuiLink,
+  EuiText,
+  EuiPopover,
+  EuiToolTip,
+  htmlIdGenerator,
+} from '@elastic/eui';
+import { keys } from '@elastic/eui';
+import { IFieldFormat } from '../../../../../../../../src/plugins/data/common';
+import { RangeTypeLens, isValidRange, isValidNumber } from './ranges';
+import { FROM_PLACEHOLDER, TO_PLACEHOLDER, TYPING_DEBOUNCE_TIME } from './constants';
+import { NewBucketButton, DragDropBuckets, DraggableBucketContainer } from '../shared_components';
+
+const generateId = htmlIdGenerator();
+
+type LocalRangeType = RangeTypeLens & { id: string };
+
+const getBetterLabel = (range: RangeTypeLens, formatter: IFieldFormat) =>
+  range.label ||
+  formatter.convert({
+    gte: isValidNumber(range.from) ? range.from : FROM_PLACEHOLDER,
+    lt: isValidNumber(range.to) ? range.to : TO_PLACEHOLDER,
+  });
+
+export const RangePopover = ({
+  range,
+  setRange,
+  Button,
+  isOpenByCreation,
+  setIsOpenByCreation,
+}: {
+  range: LocalRangeType;
+  setRange: (newRange: LocalRangeType) => void;
+  Button: React.FunctionComponent<{ onClick: MouseEventHandler }>;
+  isOpenByCreation: boolean;
+  setIsOpenByCreation: (open: boolean) => void;
+  formatter: IFieldFormat;
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [tempRange, setTempRange] = useState(range);
+
+  const saveRangeAndReset = (newRange: LocalRangeType, resetRange = false) => {
+    if (resetRange) {
+      // reset the temporary range for later use
+      setTempRange(range);
+    }
+    // send the range back to the main state
+    setRange(newRange);
+  };
+  const { from, to } = tempRange;
+
+  const lteAppendLabel = i18n.translate('xpack.lens.indexPattern.ranges.lessThanOrEqualAppend', {
+    defaultMessage: '\u2264',
+  });
+  const lteTooltipContent = i18n.translate(
+    'xpack.lens.indexPattern.ranges.lessThanOrEqualTooltip',
+    {
+      defaultMessage: 'Less than or equal to',
+    }
+  );
+  const ltPrependLabel = i18n.translate('xpack.lens.indexPattern.ranges.lessThanPrepend', {
+    defaultMessage: '\u003c',
+  });
+  const ltTooltipContent = i18n.translate('xpack.lens.indexPattern.ranges.lessThanTooltip', {
+    defaultMessage: 'Less than',
+  });
+
+  const onSubmit = () => {
+    setIsPopoverOpen(false);
+    setIsOpenByCreation(false);
+    saveRangeAndReset(tempRange, true);
+  };
+
+  return (
+    <EuiPopover
+      display="block"
+      ownFocus
+      isOpen={isOpenByCreation || isPopoverOpen}
+      closePopover={onSubmit}
+      button={
+        <Button
+          onClick={() => {
+            setIsPopoverOpen((isOpen) => !isOpen);
+            setIsOpenByCreation(false);
+          }}
+        />
+      }
+      data-test-subj="indexPattern-ranges-popover"
+    >
+      <EuiForm>
```

Comment:
Unless you're actually displaying this as a `<form>` element or submitting as one, this component doesn't actually do anything. I'd remove it.